Tools/openai_manager.py

```diff
@@ -16,31 +18,33 @@ def __new__(cls):
         return cls._instance
 
     async def process_text(self, telegram_user_id: int, text: str) -> str:
-        conv_manager = ConversationManager()
-        db_user = UserCache().get_user(telegram_user_id)
-        db_user_id = str(db_user.id)
-        
+        conv_manager, db_user_id = ConversationManager.get_user_conversation(
+            telegram_user_id)
+
+        db = next(get_db())
+        if not is_user_details_exists(db, db_user_id):
+            return process_survey_message(db_user_id, text)
+
         conv_manager.add_message(db_user_id, "user", text)
         messages = conv_manager.get_conversation_history(db_user_id)
-        
+
         response = OpenAIClient.get_instance().client.responses.create(
             model=Config.from_env().gpt_model,
             input=messages,
             instructions=CHAT_MAIN_PROMPT
         )
-        
+
         response_text = response.output_text
         conv_manager.add_message(db_user_id, "assistant", response_text)
-        
+
         return response_text
 
     async def process_image(self, telegram_user_id: int, image_path: str, text: str) -> str:
-        conv_manager = ConversationManager()
-        db_user = UserCache().get_user(telegram_user_id)
-        db_user_id = str(db_user.id)
-        
+        conv_manager, db_user_id = ConversationManager.get_user_conversation(
```

Comment:
get_conersation_history should be used instead. You can call UserCache here - it's singleton.